### ISSUE TITLE
Add a compiler option that allows defining a list of OSGI `provided` bundles

### ISSUE THREAD

**Issue by davidfestal** [ceylon/ceylon-compiler#2337]
*provided bundles* are modules that, if imported, __*should not be added*__ in the `Require-Bundle`  OSGI manifest header, because they are expected to be OSGI bundles provided by the target OSGI framework.
Typical examples of such provided modules would be the OSGI bundles of the various OSGI standard services : Instead of adding a fixed dependency to a fixed version of such a bundle, we should add a template MANIFEST.MF as a project resource, with the right `Import-Package`  header.

**Comment by FroMage** [comment]
We could also fix that with a `provided` annotation, which could be useful for the generated `pom.xml` too, but who would provide those modules at runtime anyways?

**Comment by davidfestal** [comment]
When we discussed the point, @gavinking was not OK to add this annotation afair.
Anyway, I've already implemented that under the form of an `osgi-provided-bundles` option in the compiler, and compile tool and ant task.

I'm about to push.

Something more general could be done in the future about provided modules, but for the moment it's totally sufficient in order to provide a true OSGI compatibility.
